### ISSUE TITLE
How to calculate and obtain R@5

### ISSUE THREAD

**Issue by MinionAttack** [e-bug/syncap#5]
Hi,

I'm trying to replicate some of the experiments and with _score_val_ and _bertscore_va_l scripts I can get the M, S, C, B, and BS metrics shown in Table 1. (https://arxiv.org/pdf/2101.11911.pdf)

My problem is how to calculate and obtain the R@5 metric, is there a script or function that already does it or do I have to do it manually?

Greetings.

**Comment by e-bug** [comment]
Hi,

Yes, it is all implemented. You can simply pass "recall" as one of the metrics of evaluation for `code/score.py`.

You can find ready-to-be-used evaluation files under `experiements/`. 
For example, this computes all the scores except BS for BUTD on the first heldout split: https://github.com/e-bug/syncap/blob/master/experiments/coco_heldout_1/butd/score_test.sh

**Comment by MinionAttack** [comment]
Ah, I see. But in my experiments I only have "val_xxxx.json" and not the "test_xxxx.json". Checking the available scripts I see that there is a _test.sh_. What would be the correct order to generate the experiments?

train.sh -> val.sh -> test.sh

or I can do:

train.sh -> test.sh

Becasue I can see two scripts, _score_val_ and _score_test_. If I want the test scores, can I omit the val script and its scores?

What is the correct flow?

Also, looking at the output I get using _score_val_ right now I get this:

_coco.beam_100.json_ with the scores:

```
Bleu_1: 0.738596
Bleu_2: 0.591134
Bleu_3: 0.459714
Bleu_4: 0.356414
METEOR: 0.263714
ROUGE_L: 0.570095
CIDEr: 0.966970
SPICE: 0.202339
```

And one file per pair of the heldout _recall_5.XXXX.beam_100.val_ with a dict of strings and values. Where I can get the final score or it is not output because I'm executing _score_val_ instead of _score_test_?

Thanks.

**Comment by e-bug** [comment]
You should only evaluate on the test set at the end -- after you have tuned your model (using the validation set).

To get the final results across different heldout sets etc, please see `notebooks/AverageResults.ipynb`.
The `notebooks/` folder contains more scripts to reproduce the results reported in the paper.

**Comment by MinionAttack** [comment]
Ok, thanks for the clarification. I will have a look to see if I can replicate the values in the paper.
